### ISSUE TITLE
Feature Request: Seeding models for reproducible fitting.

### ISSUE THREAD

**Issue by felipetovarhenao** [flucoma/flucoma-core#327]
### Is your feature request related to a problem? Please describe the problem.

I've noticed that for trickier machine learning tasks, the training for models such as `fluid.mlpregressor~` (and more generally the trained parameters of other objects) tend to differ greatly in accuracy/loss on each try. Not being able to fix the seed, and thus have reproducible outcomes, makes it harder to understand if and how changes in training parameters improve the model, or are just a result of a different seed.

### Describe the solution you'd like to see.

For both pedagogical and creative reasons, it would be extremely helpful to allow data objects such as `fluid.mlpregressor`/`classifier~` or `fluid.umap~`, etc., to support fixed seeding similar to the built-in random object (`0`=no fixed seed; _non-zero_ = fixed seed). Considering this is available in libraries like `pytorch`, it would seem like a natural feature to add.

### Describe alternatives you've considered

The only fix is at this moment is writing the JSON model to file, but it still doesn't solve the educational value of reproducing the training process.

### Additional context

A basic example would be:

with fixed seed
```
[fluid.mlpregressor~ @seed 1]
```

without fixed seed (default for backwards compatibility)

```
[fluid.mlpregressor~ @seed 0]
```

It might require an additional method/message to reset seed

**Comment by felipetovarhenao** [comment]
Hi @tremblap I think this could work as a temporary workaround but there's still some friction I see (unless I'm misunderstanding your approach):

1. If I follow correctly, seeding the model would always require creating a dummy input/output set pair, fitting it, and writing the parameters to a JSON file, which itself could only be used for models with the exact same layer configuration. Another alternative in Max would be to manually construct a `dict` and randomize the parameters with a `[random]` object, but handling nested arrays, which would be required for the `"weights"` parameter, is a bit of headache in Max dictionaries.
2. Generating the initial seed would still not be reproducible, which would require always sharing code + JSON to reproduce results across machines.

It just seems like adding a new `@seed` argument would make for a much smoother experience and it would still allow for the kind of seeding you are suggesting, which is necessary for supporting pre-trained models.

**Comment by tremblap** [comment]
ok @felipetovarhenao @weefuzzy has implemented it for all random-based objects (except kmeans and skmeans which will arrive eventually with their rewrite) and transient stuff (because it needs too much work for its current state and is in deep refactor)

feel free to test the nightlies compiled today, and please confirm here that it works as you'd like and we'll close the ticket

**Comment by felipetovarhenao** [comment]
That's awesome, @tremblap and @weefuzzy ! Thanks to both for including this feature! @tremblap could you clarify where the nightlies are hosted? I'll try to test them sometime today and get back to you.

**Comment by felipetovarhenao** [comment]
Nevermind, found them:)

**Comment by tremblap** [comment]
did you try reseeding? clear, reset, seed, fit, dump?

**Comment by felipetovarhenao** [comment]
@weefuzzy Did a few quick tests and so far it does seem to work! I should mention that `@seed 0`, just like `@seed -1`, behaves as no seed—I prefer it that way, but since you had mentioned it was only meant to be `-1` I thought I should flag it just in case it wasn't intentional.

I will run more substantial tests later in the day with the other objects and report back.
